code/src/components/Text.js

```diff
@@ -0,0 +1,24 @@
+import React from 'react';
+
+import '../style/input-box.css';
+
+export const Text = ({ name, setName, askQuestion, handleNameChange }) => {
+
+  return(
+    <div>
+      <label htmlFor="userName"><h2>{askQuestion}</h2></label>
+        <input
+          id="userName"
+          className="input-text"
+          onKeyPress={event => {event.key === 'Enter' && event.preventDefault();}}
+          type="text"
+          onChange={handleNameChange}
+          value={name}
+          placeholder="Enter name here..."
+          required
+          minLength="2"
```

Comment:
It seems that the validation minLenght here does not work when I try typing 1 letter. I have the same problem and do not know how to solve it either.

Reply:
I believe this is because the button clicked is just a button and not on submit. This type of validation only works onSubmit. I have removed it from my code as it isn't doing anything.